lib/task_runner.rb

```diff
@@ -0,0 +1,39 @@
+class TaskRunner
+  class << self
+    def run_build(request)
```

Comment:
It's difficult to test System calls. You could mock them, and verify that they get called, but you wouldn't test that the commands work, because that would be test those tools, which should have testing.